sim/battle.ts

```diff
@@ -1127,42 +1127,49 @@ export class Battle extends Dex.ModdedDex {
 			let maxTeamSize = teamLengthData && teamLengthData.battle;
 			this.add('teampreview' + (maxTeamSize ? '|' + maxTeamSize : ''));
 			if (!maxTeamSize) maxTeamSize = 6;
-			this.p1.maxTeamSize = maxTeamSize;
-			this.p2.maxTeamSize = maxTeamSize;
-			this.p1.currentRequest = 'teampreview';
-			p1request = {teamPreview: true, maxTeamSize, side: this.p1.getRequestData()};
-			this.p2.currentRequest = 'teampreview';
-			p2request = {teamPreview: true, maxTeamSize, side: this.p2.getRequestData()};
+			for (let i = 0; i < this.sides.length; i++) {
+				let side = this.sides[i];
+				side.maxTeamSize = maxTeamSize;
+				side.currentRequest = 'teampreview';
+				requests[i] = {teamPreview: true, maxTeamSize, side: side.getRequestData()};
+			}
 			break;
 
 		default: {
-			this.p1.currentRequest = 'move';
-			let activeData = this.p1.active.map(pokemon => pokemon && pokemon.getRequestData());
-			p1request = {active: activeData, side: this.p1.getRequestData()};
-
-			this.p2.currentRequest = 'move';
-			activeData = this.p2.active.map(pokemon => pokemon && pokemon.getRequestData());
-			p2request = {active: activeData, side: this.p2.getRequestData()};
+			for (let i = 0; i < this.sides.length; i++) {
+				let side = this.sides[i];
+				side.currentRequest = 'move';
+				let activeData = side.active.map(pokemon => pokemon && pokemon.getRequestData());
+				requests[i] = {active: activeData, side: side.getRequestData()};
+			}
 			break;
 		}
 		}
 
-		if (p1request) {
-			if (!this.supportCancel || !p2request) p1request.noCancel = true;
-			this.p1.emitRequest(p1request);
-		} else {
-			this.p1.emitRequest({wait: true, side: this.p1.getRequestData()});
+		let allRequestsMade = true;
+		for (const request of requests) {
+			if (!request) allRequestsMade = false;
```

Comment:
Without my adjustment for `requests`, this would have made this false for all single battles...

Reply:
double and triple battles too